stdlib/LinearAlgebra/src/generic.jl

```diff
@@ -478,7 +478,7 @@ function generic_norm1(x)
     (v, s) = iterate(x)::Tuple
     av = float(norm(v))
     T = typeof(av)
-    sum::promote_type(Float64, T) = av
+    sum = av*1.0 # promotes to at least Float64 with * 1.0
```

Comment:
```suggestion
    sum = float(av)
```
Or are there cases where `float` will create a `Float32`?

Reply:
Of course there are, `float(::Float32)`. Just ignore this.